pkg/block/fetcher.go

```diff
@@ -353,19 +473,16 @@ func (f *BaseFetcher) fetchMetadata(ctx context.Context) (interface{}, error) {
 	})
 
 	if err := eg.Wait(); err != nil {
-		return nil, errors.Wrap(err, "BaseFetcher: iter bucket")
+		return resp, errors.Wrap(err, "BaseFetcher: iter bucket")
 	}
 
 	if len(resp.metaErrs) > 0 {
 		return resp, nil
 	}
 
 	// Only for complete view of blocks update the cache.
-	cached := make(map[ulid.ULID]*metadata.Meta, len(resp.metas))
-	for id, m := range resp.metas {
-		cached[id] = m
-	}
-	f.cached = cached
+	f.cached = resp.metasCopy()
+	f.marks = resp.marks // no need to copy, as it's not going to be modified
```

Comment:
Given deletion marks are relatively small (compared to metas), could just be more future-proof always copying them?